shared/django_apps/core/managers.py

```diff
@@ -284,7 +285,7 @@ def with_oldest_commit_at(self):
     def get_or_create_from_git_repo(self, git_repo, owner):
         from shared.django_apps.codecov_auth.models import Owner
 
-        repo, created = self.get_or_create(
```

Comment:
I'm not sure how this fixes it, so the unique constraints are the triplet of `service_id`, `name`, and `author `right? And I'm guessing the initial error happens when `get_or_create` is called with the same above triplet but different `branch` or `private` values. It doesn't recognize these 5 args being in the DB so it tries to create a new one, but failing from the unique constraints. This is what's currently happening right?

Now with `update_or_create`, similarly it doesn't find these 5 args in the DB so it won't update it would do create leading to the same error no?

Reply:
I'm thinking the way to do it is one of these 2:
1. Ignore changes on subsequent changes to `branch` and `private` columns
```
repo, created = self.get_or_create(
    author=owner,
    service_id=git_repo.get("service_id") or git_repo.get("id"),
    name=git_repo["name"],
    defaults={
        "branch": git_repo.get("branch") or git_repo.get("default_branch") or "main",
        "private": git_repo["private"],
    }
)
```
1. Updates changes on subsequent changes to `branch` and `private` columns
```
repo, created = self.update_or_create(
    author=owner,
    service_id=git_repo.get("service_id") or git_repo.get("id"),
    name=git_repo["name"],
    defaults={
        "branch": git_repo.get("branch") or git_repo.get("default_branch") or "main",
        "private": git_repo["private"],
    }
)
```

Reply:
Actually that's not gonna work because the unique constraints are 2 pairs and `get_or_update` is 1 triplet. So if you had in the DB `(author=codecov, service_id=1, name=gazebo)`
and call with params:
`(author=codecov, service_id=2, name=gazebo)`
a unique constraint error would occur as well because (author, name) constraint fail.